### PR TITLE
Sensei Update

### PR DESCRIPTION
**Pull request by aumu1031**
Imgur gallery: https://imgur.com/a/FqdeChM (minor styling tweaks since)
Test cases used:
```
All Players:
	VillageHQ
		- Sensei tab to nav: display sensei detail listing with link to sensei/student profiles, restricted by village and sorted by graduated count
Akademi/Genin:
	Profile
		- Not Student: Display section with link to VillageHQ->Sensei
		- Is Student: Display section with sensei details, link to sensei/student profiles
	VillageHQ
		- Display informational section
			- Has active application: display grid with link to sensei, link to revoke application
		- Actions:
			- Apply: check if already have sensei, rank, if target player is sensei, village, accepting students, error if student roster full or duplicate application, create notification for sensei
			- Cancel Application
			- Leave Sensei
	Members
		- If user is a sensei, same village, accepting applications, link to apply (villageHQ)
		- If user is player's sensei, link to abandon sensei (villageHQ)
	Premium
		- Restrict village change if student
		- Delete active student applications for user
	Level-Up
		- Clear sensei_id from user table
		- Remove user_id from sensei table student_list
		- Increment sensei graduated count
Jonin:
	Profile
		- Is Sensei: Display section with sensei details, link to sensei/student profiles and villageHQ (if student available)
		- Actions:
			- Update specialization (dropdown, Tai/Nin/Gen)
			- Update student message (length 500, html parsed)
			- Update recruitment setting - accept students (checkbox toggle)
			- Update recruitment message (length 100, html parsed)
	VillageHQ
		- Display informational section
			- If not sensei: link to exam
			- If sensei: link to resign -> link to confirm
			- If has active application: display grid with link to student, links to approve/deny
		- Actions:
			- Take Exam: check if already sensei, is Jonin, is >= 75, has 5 jutsu mastered
			- Submit Exam: check if already sensei, is Jonin, is >= 75, has 5 jutsu mastered, has correct answers
			- Resign / Confirm: check if sensei
			- Kick Student
			- Accept Application: clear applications if student count = 3
			- Deny Application
	Members
		- If user is student, link to kick (villageHQ)
	Premium
		- Restrict village change if has students
		- Delete active student applications for user
Moderators:
	Village HQ
		- Swap between village views, can clear any recruitment message
```

### CODE REVIEW

**Comment by aumu1031** [comment]
Still going through a full run of testing and tweaks, just wanted to get this out there for review.

**Comment by aumu1031** [comment]
Passed all listed test cases after review.

File: db/migrations/20230512113441_sensei_migration.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{
+    /**
+     * Migrate Up.
+     */
+    public function up(): void
+    {
+        // sensei table
+        $this->execute("
+            CREATE TABLE `sensei` (
+                `sensei_id` int(11) NOT NULL,
```
Comment (levimeahan) [inline db/migrations/20230512113441_sensei_migration.php:16]: To make this clear, could we change this field to `sensei_user_id` or just `user_id`? It's a little weird reading through the rest of the code and trying to understand if there's a case where `sensei_id` would not match the user's ID

File: classes/SenseiManager.php [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        $system->query("UPDATE `sensei` SET `students` = '{$students}' WHERE `sensei_id` = '{$sensei_id}'");
+        $system->query("UPDATE `users` SET `sensei_id` = '0' WHERE `user_id` = '{$student_id}'");
+        $system->query("COMMIT;");
+        if ($system->db_last_num_rows > 0) {
+            $db_modified = true;
+        }
+        return $db_modified;
+    }
+
+    public static function addSensei(int $sensei_id, $specialization, System $system): bool {
```
Comment (levimeahan) [inline classes/SenseiManager.php:176]: Similarly here, if you're reading this in isolation it's not clear what the "sensei ID" is supposed to be. If we name it `sensei_user_id` or just `user_id` it makes it a little more clear

**Review by levimeahan — APPROVED** [review]
One major comment but other than that this looks good, let's test it out!

File: db/migrations/20230512113441_sensei_migration.php (hunk 1 above)
Comment (aumu1031) [reply db/migrations/20230512113441_sensei_migration.php:16]: For context:
When a user has `sensei_id` == `user_id` it indicates they are the sensei.
When a user has `sensei_id` != `user_id` it indicates they are a student.
If `sensei_id` == NULL || 0 they are neither.

The sensei table uses `sensei_id` as the primary key to be clear it corresponds to the `sensei_id` in the user table.

I'm fine changing it to `sensei_user_id`, just want to keep it consistent between the user and sensei table. 
Thoughts?

**Pull request merged**